### PR TITLE
add vtbwife

### PR DESCRIPTION
**Pull request by lianhong2758**

### CODE REVIEW

File: plugin/vtbwife/vtb.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		PublicDataFolder: "VtbWife",
+	})
+	var keys []string
+	engine.OnRegex(`^抽(vtb|VTB)(老婆)?$`, fcext.DoOnceOnSuccess(
+		func(ctx *zero.Ctx) bool {
+			content, err := engine.GetLazyData("wife_list.txt", false)
+			if err != nil {
+				panic(err)
+			}
+			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:34]: 没用的删干净

File: plugin/vtbwife/vtb.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+			if err != nil {
+				log.Println("[vtbwife]读取vtbwife数据文件失败: ", err)
+				return false
+			}*/
+			// 将文件内容转换为单词
+			keys = strings.Split(string(content), "\n")
+			log.Println("[vtbwife]加载", len(keys), "位wife数据...")
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:41]:
```go
log.Infoln("[vtbwife]加载", len(keys), "位 vtb ...")
```

File: plugin/vtbwife/vtb.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+			return true
+		})).SetBlock(true).Handle(func(ctx *zero.Ctx) {
+		var key, u, b string
+		var ok bool
+		for i := 0; i < 3; i++ {
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:46]: for i := 0; i < 3 && !ok; i++

File: plugin/vtbwife/vtb.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i]
+			u, b, ok = geturl(key)
+			if !ok {
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:49]: remove

File: plugin/vtbwife/vtb.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+				continue
+			}
+			break
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:52]: remove

File: plugin/vtbwife/vtb.go [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+		}
+		if !ok {
+			ctx.SendChain(message.Text("-获取图片链接失败"))
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:55]: 这个诡异的`-`前缀是什么？全删了。

File: plugin/vtbwife/vtb.go [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+			return
+		}
+		img, err := web.GetData(u)
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:58]: 莲宝笨笨，直接把url放message.Image()不好吗？

File: plugin/vtbwife/vtb.go [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+	doc.Find("style").Remove()
+	doc.Find("script").Remove()
+	doc.Find(".fans-medal-level").Remove()
+	var b []string
+	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) {
+		b = append(b, strings.TrimSpace(s.Text()))
+	})
+	var k int
+	for kk, vv := range b {
+		if strings.TrimSpace(vv) == "基本资料" || strings.TrimSpace(vv) == "基本信息" || strings.TrimSpace(vv) == "名字" || strings.TrimSpace(vv) == "名称" {
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:93]: 你还真调用这么多次strings.TrimSpace啊🤣👉

File: plugin/vtbwife/vtb.go [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+	u, ok = doc.Find(".infobox-image").Attr("src") // class加.
+	doc.Find("style").Remove()
+	doc.Find("script").Remove()
+	doc.Find(".fans-medal-level").Remove()
+	var b []string
+	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) {
+		b = append(b, strings.TrimSpace(s.Text()))
+	})
+	var k int
+	for kk, vv := range b {
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:92]: kk, vv：罚你重新起名😠

File: plugin/vtbwife/vtb.go [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+	if err != nil {
+		return "", "", false
+	}
+	u, ok = doc.Find(".infobox-image").Attr("src") // class加.
+	doc.Find("style").Remove()
+	doc.Find("script").Remove()
+	doc.Find(".fans-medal-level").Remove()
+	var b []string
+	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) {
+		b = append(b, strings.TrimSpace(s.Text()))
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:89]: 有没有一种可能，就是说这里就是一个循环，你可以在这里把下面的事全干了

File: plugin/vtbwife/vtb.go [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+	})
+	var k int
+	for kk, vv := range b {
+		if strings.TrimSpace(vv) == "基本资料" || strings.TrimSpace(vv) == "基本信息" || strings.TrimSpace(vv) == "名字" || strings.TrimSpace(vv) == "名称" {
+			k = kk + 1
+			break
+		}
+	}
+	if k != 0 {
+		brief = b[k-1] + "\n"
```
Comment (fumiama) [inline plugin/vtbwife/vtb.go:99]: 用strings.Builder

**Review by fumiama — CHANGES_REQUESTED** [review]

**Comment by lianhong2758** [comment]
TD




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年4月17日(星期一) 晚上8:11
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [FloatTech/ZeroBot-Plugin] add vtbwife (PR #672)





 
@fumiama requested changes on this pull request.
 
 
In go.mod:
 &gt; @@ -13,6 +13,7 @@ require (  	github.com/FloatTech/ttl v0.0.0-20220715042055-15612be72f5b  	github.com/FloatTech/zbpctrl v1.5.3-0.20230411055601-7cdff751dd67  	github.com/FloatTech/zbputils v1.6.2-0.20230411114057-4c71136244b2 +	github.com/PuerkitoBio/goquery v1.5.1  
莲宝快用已有的htmlquery
 
 
In plugin/vtbwife/vtb.go:
 &gt; + +func init() { // 插件主体 +	engine := control.Register("vtbwife", &amp;ctrl.Options[*zero.Ctx]{ +		DisableOnDefault: false, +		Brief:            "抽vtb老婆", +		Help:             "- 抽vtb(老婆)", +		PublicDataFolder: "VtbWife", +	}) +	var keys []string +	engine.OnRegex(`^抽(vtb|VTB)(老婆)?$`, fcext.DoOnceOnSuccess( +		func(ctx *zero.Ctx) bool { +			content, err := engine.GetLazyData("wife_list.txt", false) +			if err != nil { +				panic(err) +			} +			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界  
没用的删干净
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	}) +	var keys []string +	engine.OnRegex(`^抽(vtb|VTB)(老婆)?$`, fcext.DoOnceOnSuccess( +		func(ctx *zero.Ctx) bool { +			content, err := engine.GetLazyData("wife_list.txt", false) +			if err != nil { +				panic(err) +			} +			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界 +			if err != nil { +				log.Println("[vtbwife]读取vtbwife数据文件失败: ", err) +				return false +			}*/ +			// 将文件内容转换为单词 +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...")  log.Infoln("[vtbwife]加载", len(keys), "位 vtb ...")
 
 
In plugin/vtbwife/vtb.go:
 &gt; +			if err != nil { +				panic(err) +			} +			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界 +			if err != nil { +				log.Println("[vtbwife]读取vtbwife数据文件失败: ", err) +				return false +			}*/ +			// 将文件内容转换为单词 +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...") +			return true +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ {  
for i := 0; i < 3 &amp;&amp; !ok; i++
 
 
In plugin/vtbwife/vtb.go:
 &gt; +			/*content, err := os.ReadFile(engine.DataFolder() + "wife_list.txt") // 779分界 +			if err != nil { +				log.Println("[vtbwife]读取vtbwife数据文件失败: ", err) +				return false +			}*/ +			// 将文件内容转换为单词 +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...") +			return true +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ { +			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i] +			u, b, ok = geturl(key) +			if !ok {  
remove
 
 
In plugin/vtbwife/vtb.go:
 &gt; +				return false +			}*/ +			// 将文件内容转换为单词 +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...") +			return true +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ { +			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i] +			u, b, ok = geturl(key) +			if !ok { +				continue +			} +			break  
remove
 
 
In plugin/vtbwife/vtb.go:
 &gt; +			keys = strings.Split(string(content), "\n") +			log.Println("[vtbwife]加载", len(keys), "位wife数据...") +			return true +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ { +			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i] +			u, b, ok = geturl(key) +			if !ok { +				continue +			} +			break +		} +		if !ok { +			ctx.SendChain(message.Text("-获取图片链接失败"))  
这个诡异的-前缀是什么？全删了。
 
 
In plugin/vtbwife/vtb.go:
 &gt; +		})).SetBlock(true).Handle(func(ctx *zero.Ctx) { +		var key, u, b string +		var ok bool +		for i := 0; i < 3; i++ { +			key = keys[fcext.RandSenderPerDayN(ctx.Event.UserID, len(keys))+i] +			u, b, ok = geturl(key) +			if !ok { +				continue +			} +			break +		} +		if !ok { +			ctx.SendChain(message.Text("-获取图片链接失败")) +			return +		} +		img, err := web.GetData(u)  
莲宝笨笨，直接把url放message.Image()不好吗？
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	// 使用goquery解析网页内容 +	doc, err := goquery.NewDocumentFromReader(resp.Body) +	if err != nil { +		return "", "", false +	} +	u, ok = doc.Find(".infobox-image").Attr("src") // class加. +	doc.Find("style").Remove() +	doc.Find("script").Remove() +	doc.Find(".fans-medal-level").Remove() +	var b []string +	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) { +		b = append(b, strings.TrimSpace(s.Text())) +	}) +	var k int +	for kk, vv := range b { +		if strings.TrimSpace(vv) == "基本资料" || strings.TrimSpace(vv) == "基本信息" || strings.TrimSpace(vv) == "名字" || strings.TrimSpace(vv) == "名称" {  
你还真调用这么多次strings.TrimSpace啊🤣👉
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	defer resp.Body.Close() +	// 使用goquery解析网页内容 +	doc, err := goquery.NewDocumentFromReader(resp.Body) +	if err != nil { +		return "", "", false +	} +	u, ok = doc.Find(".infobox-image").Attr("src") // class加. +	doc.Find("style").Remove() +	doc.Find("script").Remove() +	doc.Find(".fans-medal-level").Remove() +	var b []string +	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) { +		b = append(b, strings.TrimSpace(s.Text())) +	}) +	var k int +	for kk, vv := range b {  
kk, vv：罚你重新起名😠
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	if err != nil { +		return "", "", false +	} +	defer resp.Body.Close() +	// 使用goquery解析网页内容 +	doc, err := goquery.NewDocumentFromReader(resp.Body) +	if err != nil { +		return "", "", false +	} +	u, ok = doc.Find(".infobox-image").Attr("src") // class加. +	doc.Find("style").Remove() +	doc.Find("script").Remove() +	doc.Find(".fans-medal-level").Remove() +	var b []string +	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) { +		b = append(b, strings.TrimSpace(s.Text()))  
有没有一种可能，就是说这里就是一个循环，你可以在这里把下面的事全干了
 
 
In plugin/vtbwife/vtb.go:
 &gt; +	doc.Find("style").Remove() +	doc.Find("script").Remove() +	doc.Find(".fans-medal-level").Remove() +	var b []string +	doc.Find(".moe-infobox").Find("tr").Each(func(i int, s *goquery.Selection) { +		b = append(b, strings.TrimSpace(s.Text())) +	}) +	var k int +	for kk, vv := range b { +		if strings.TrimSpace(vv) == "基本资料" || strings.TrimSpace(vv) == "基本信息" || strings.TrimSpace(vv) == "名字" || strings.TrimSpace(vv) == "名称" { +			k = kk + 1 +			break +		} +	} +	if k != 0 { +		brief = b[k-1] + "\n"  
用strings.Builder

**Comment by lianhong2758** [comment]
????




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年4月17日(星期一) 晚上8:23
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [FloatTech/ZeroBot-Plugin] add vtbwife (PR #672)





 
close

File: plugin/vtbwife/vtb.go (hunk 10 above)
Comment (lianhong2758) [reply plugin/vtbwife/vtb.go:89]: 要找开头,这个怎么在这里实现呢,如果没找到则全部写进去

Comment (lianhong2758) [reply plugin/vtbwife/vtb.go:89]: 要找开头,这个怎么在这里实现呢,如果没找到则全部写进去

Comment (fumiama) [reply plugin/vtbwife/vtb.go:73]: 你下面怎么循环的，把它搬上来呗

File: go.mod [hunk 12]
```diff
@@ -13,6 +13,7 @@ require (
 	github.com/FloatTech/ttl v0.0.0-20220715042055-15612be72f5b
 	github.com/FloatTech/zbpctrl v1.5.3-0.20230411055601-7cdff751dd67
 	github.com/FloatTech/zbputils v1.6.2-0.20230411114057-4c71136244b2
+	github.com/PuerkitoBio/goquery v1.5.1
```
[the remark this answers is not in the document]
Comment (fumiama) [reply go.mod:16]: 自己百度

**Review by fumiama — CHANGES_REQUESTED** [review]

File: go.mod (hunk 12 above)
[the remark this answers is not in the document]
Comment (guohuiyuan) [reply go.mod:16]: 用gpt

**Comment by medicago087** [comment]
莲宝该动一动了

**Comment by lianhong2758** [comment]
苜蓿子快帮我改改，求求了



---Original---
From: ***@***.***&gt;
Date: Fri, Apr 21, 2023 16:25 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [FloatTech/ZeroBot-Plugin] add vtbwife (PR #672)




 
莲宝该动一动了

**Comment by medicago087** [comment]
加油，我能力越小责任越小了

File: go.mod (hunk 12 above)
[the remark this answers is not in the document]
Comment (lianhong2758) [reply go.mod]: 很稳定,两天测试没有问题,还可以统一数据

**Review by fumiama — CHANGES_REQUESTED** [review]
有点气了😠你整扣扣米可以，谁让你搞这个了？zbpdata都合了，不用说不过去吧，

**Comment by guohuiyuan** [comment]
这不直接dds攻击，搞挂它

**Comment by lianhong2758** [comment]
> 有点气了 😠 你整扣扣米可以，谁让你搞这个了？zbpdata都合了，不用说不过去吧，

呜呜呜

**Pull request closed without merging**